R/cache.R

```diff
@@ -70,9 +70,7 @@ cache_boards <- function() {
 }
 
 cache_versions <- function() {
-  pins <- fs::dir_ls(cache_boards(), type = "directory")
-  versions <- fs::dir_ls(pins, type = "directory")
-
+  versions <- fs::dir_ls(cache_boards(), recurse = 2, type = "directory")
```

Comment:
Do you know why this behaves differently? I wouldn't have expected this code to give a different result, so something about my understanding of `dir_ls()` is flawed.

Reply:
I think it's that `recurse = 2` goes down 2 levels and the old code just went down one:

``` r
fs::dir_ls(type = "directory")
#> R         blog      inst      man       revdep    tests     tools     vignettes
fs::dir_ls(type = "directory", recurse = 1)
#> R                  blog               blog/pins-0-2-0    blog/pins-0-3-0    
#> blog/pins-0-4-0    inst               inst/preview       inst/secret        
#> man                man/figures        revdep             tests              
#> tests/testthat     tools              vignettes          vignettes/articles
fs::dir_ls(type = "directory", recurse = 2)
#> R                                blog                             
#> blog/pins-0-2-0                  blog/pins-0-2-0/images           
#> blog/pins-0-3-0                  blog/pins-0-3-0/images           
#> blog/pins-0-4-0                  blog/pins-0-4-0/images           
#> inst                             inst/preview                     
#> inst/preview/highlight.js-9.15.9 inst/preview/pagedtable-1.1      
#> inst/secret                      man                              
#> man/figures                      revdep                           
#> tests                            tests/testthat                   
#> tests/testthat/_snaps            tests/testthat/datatxt           
#> tests/testthat/files             tests/testthat/pin-board         
#> tests/testthat/pin-files         tests/testthat/pin-rds           
#> tools                            vignettes                        
#> vignettes/articles
```

<sup>Created on 2023-04-19 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

Reply:
Oooh, duh.